packages/indexer-agent/src/agent.ts

```diff
@@ -339,12 +339,16 @@ class Agent {
         }
 
         try {
-          await this.reconcileDeployments(activeDeployments, targetDeployments)
+          await this.reconcileDeployments(
+            activeDeployments,
+            targetDeployments,
+            activeAllocations,
+          )
 
           // Reconcile allocations
           await this.reconcileAllocations(
             activeAllocations,
-            targetDeployments,
+            activeDeployments,
             indexingRules,
             currentEpoch.toNumber(),
             currentEpochStartBlock,
```

Comment:
That's ok though right? It'll pick it up in the next loop.

Reply:
True. Maybe that extra time is actually good, in case a subgraph fails right after starting.

Can you add a comment between these two `reconcile*` functions that deployments that start indexing above will only be allocated towards in the next loop?

Reply:
Yeah my thoughts exactly, the gap between loops will make it safer because it will give it time to try syncing it first. 

I'll add a comment explaining this.